tests/integration/models/device.spec.coffee

```diff
@@ -1028,6 +1028,32 @@ describe 'Device Model', ->
 					m.chai.expect(_.find(result, { name: 'A' }).value).equal('a')
 					m.chai.expect(_.find(result, { name: 'B' }).value).equal('b')
 
+		describe 'balena.models.device.getSupervisorTargetState()', ->
+
+			it 'should be rejected if the device does not exist', ->
+				promise = balena.models.device.getSupervisorTargetState('asdfghjkl')
+				m.chai.expect(promise).to.be.rejectedWith('Device not found: asdfghjkl')
+
+			it 'should reflect the device\'s name', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) =>
+					m.chai.expect(state.local.name).to.be.a('string')
+					m.chai.expect(state.local.name).to.equal(@device.device_name)
+
+			it 'should reflect the device\'s apps', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) =>
+					m.chai.expect(state.local.apps).to.be.an('object')
+					m.chai.expect(state.local.apps[@application.id].name).to.equal(@application.app_name)
+					m.chai.expect(state.local.apps[@application.id].name).to.be.a('string')
+					m.chai.expect(state.local.apps[@application.id].services).to.be.an('object')
+					m.chai.expect(state.local.apps[@application.id].volumes).to.be.an('object')
+					m.chai.expect(state.local.apps[@application.id].networks).to.be.an('object')
+
+			it 'should reflect the device\'s config', ->
+				balena.models.device.getSupervisorTargetState(@device.id).then (state) ->
```

Comment:
Since this and the above two tests are doing the same call & the state endpoint is known to be slow, let's combine them to same some seconds in the test suit execution time & be more kind with the API.
```
        ✓ should reflect the device's name (1161ms)
        ✓ should reflect the device's apps (1125ms)
        ✓ should reflect the device's config (1138ms)
```

Reply:
great catch, i've combined these tests and commented the different sections accordingly.